ex02.exs

```diff
@@ -1,11 +1,39 @@
 
 defmodule Ex02 do
 
+  # Functions for second test
+
+  def new_counter(value) do
+    {:ok, counter} = Agent.start(fn -> value end)
+    counter
+  end
+
+  def next_value(counter) do
+    Agent.get_and_update(counter, &{ &1, &1 + 1 })
+  end
+
+  # Functions for third test
+
+  def new_global_counter(value \\ 0) do
+    send self(), Agent.start(fn -> value end)
+  end
+
+  def global_next_value do
+    receive do
+      {:ok, counter} ->
+        new_value = Agent.get(counter, &(&1))
+        new_global_counter(new_value + 1)
```

Comment:
## -4

Although this works, what you're doing here is creating a new process each time `global_next_value` is called. These processes never go away.

You were supposed to use `Agent.get_and_update` to change the value inside the agent itself.